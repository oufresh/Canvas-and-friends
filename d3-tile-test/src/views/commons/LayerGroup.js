//@flow
import React from "react";
import { transformToString } from "../transforms";
import { type ZoomTransform } from "../../modules/maps";

type LayerGroupPropsT = {
  transform: ZoomTransform,
  svgContent: string
};

const createSvgMarkup = (svg: string) => {
  return { __html: svg };
};

const LayerGroup = (props: LayerGroupPropsT) => {
  const { transform } = props;
  const content = createSvgMarkup(props.svgContent);
  const strTr = transform
    ? transformToString(
        transform.transformX ? transform.transformX : 0,
        transform.transformY ? transform.transformY : 0,
        transform.currentExpScale ? transform.currentExpScale : 1
      )
    : "";
  return <g transform={strTr} dangerouslySetInnerHTML={content} />;
};

export { LayerGroup };
