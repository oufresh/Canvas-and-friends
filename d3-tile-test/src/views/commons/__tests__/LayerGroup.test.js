import React from "react";
import { shallow } from "enzyme";
import { LayerGroup } from "../LayerGroup";

describe("LayerGroup", () => {
  it("renders without a crash ", () => {
    const transform = {
      transformX: 0,
      transformY: 0,
      currentExpScale: 1
    };
    const component = shallow(
      <LayerGroup
        transform={transform}
        svgContent={'<rect x="0" y="0" width="100" height="100" />'}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
