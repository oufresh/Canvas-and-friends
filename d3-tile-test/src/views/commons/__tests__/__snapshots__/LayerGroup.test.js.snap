// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerGroup renders without a crash  1`] = `
<g
  dangerouslySetInnerHTML={
    Object {
      "__html": "<rect x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" />",
    }
  }
  transform="translate(0,0) scale(1)"
/>
`;
